tests/main/interfaces-wayland/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: Ensure that the wayland interface works
+
+# Only test on classic Ubuntu amd64 systems that have wayland
+systems: [ ubuntu-1*-*64 ]
+
+prepare: |
+    . $TESTSLIB/pkgdb.sh
+    snap install --edge test-wayland
+
+execute: |
+    CONNECTED_PATTERN=":wayland +test-wayland"
+    DISCONNECTED_PATTERN="\- +test-wayland:wayland"
+
+    echo "Create XDG_RUNTIME_DIR=/run/user/12345"
+    mkdir -p /run/user/12345 || true
+    chmod 700 /run/user/12345
+    chown test:test /run/user/12345
+
+    echo "Start weston compositor under test user"
+    XDG_RUNTIME_DIR=/run/user/12345 su -p -c "weston --backend=headless-backend.so" test &
+
+    echo "===================================="
+
+    echo "When the plug is connected"
+    snap connect test-wayland:wayland
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    echo "Then the snap command under the test user is able connect to the wayland socket"
+    XDG_RUNTIME_DIR=/run/user/12345 su -p -l -c test-wayland test | MATCH wl_compositor
+
+    if [ "$(snap debug confinement)" = "partial" ] ; then
+        exit 0
+    fi
+
+    echo "===================================="
+
+    echo "When the plug is disconnected"
+    snap disconnect test-wayland:wayland
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "Then the snap command is not able to connect to the wayland socket"
+    if XDG_RUNTIME_DIR=/run/user/12345 su -p -l -c test-wayland test; then
+        echo "Expected error with plug disconnected"
+
+        echo "Stop weston compositor"
+        /usr/bin/killall -9 /usr/bin/weston || true
```

Comment:
I think we want this also in `restore:` to ensure if anything fails during the test we do proper cleanup.

Reply:
Note the comment below:
```
# If this is in 'restore', it hangs the test
```

Reply:
I've decided to add the killall to restore but leave it at the end of execute to prevent the hang in the normal case.